src/Mod/Draft/draftutils/make_utils.py

```diff
@@ -0,0 +1,108 @@
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   FreeCAD is distributed in the hope that it will be useful,            *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with FreeCAD; if not, write to the Free Software        *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+"""Provides utility functions for make_polarray and make_circulararray."""
+## @package make_utils
+# \ingroup DRAFT
+# \brief This module provides functions for make_polarray & make_circulararray.
+
+import FreeCAD as App
+import Part
+
+import draftutils.utils as utils
+
+from draftutils.messages import _msg, _err
+from draftutils.translate import _tr
+
+
+def make_polcirc_shared(_name, center, axis_object, axis_edge):
+    """Typecheck center, axis_object and axis_edge. Return axis_reference if
+    axis_object and axis_edge (optional) are set and correct. Otherwise return
+    None.
+
+    This is the shared functionality of make_polar_array and make_circulararray.
+    """
+    CORRECT = True
+    axis_reference = None
+
+    _msg("center: {}".format(center))
+    try:
+        utils.type_check([(center, App.Vector)], name=_name)
+    except TypeError:
+        _err(_tr("Wrong input: must be a vector."))
+        return not CORRECT, axis_reference
+
+    _msg("axis_object: {}".format(axis_object))
+    _msg("axis_edge: {}".format(axis_edge))
+    axis = axis_object
+    if axis_object and isinstance(axis_object, str):
+        found, axis = utils.find_object(axis_object,
+                                        doc=App.activeDocument())
+        if not found:
+            _err(_tr(
+                "Wrong input. Given axis_name does not refer to an "
+                "existing DocumentObject."))
+            return not CORRECT, axis_reference
+    if axis:
+        axis_edge_name = axis_edge
+        if axis_edge:
+            if isinstance(axis_edge, str):
+                print(axis)
+                print(axis_edge)
+                edge_object = axis.getSubObject(axis_edge)
+            elif isinstance(axis_edge, int):
+                axis_edge_name = "Edge" + str(axis_edge)
+                edge_object = axis.getSubObject(axis_edge_name)
```

Comment:
Have you made a decision on which should be the input for `axis_edge` here? I think if we restrict the input to being an `int`, it is better than passing a `str`. The `int` is easier to handle and to convert to an `EdgeN`. The minimum value of `int` should be `1`, and shouldn't allow negative values.

The problem with accepting a string is that the data is more fixed, while an `int` you can reinterpret in different ways if the internal structure changes. I'm currently working on the `Draft Dimension` code, and there the `make_functions` accept `int` to define the edges to measure. In that case the minimum value is `0`, but I'm changing it so that the minimum value is `1`.

Reply:
I thought the more options the better, but you're talking about future compatibility of files created with old FreeCAD versions? Because in the `PropertySubLinkGlobal` the edge is also saved as String so I think this won't make a difference? 

I noticed that there are cases where the edges are named differently, e.g. it's called `Line`  for an `DatumLine`. There would also be the option to access the edges via `Object.Shape.Edges` but I am not sure how the Order of the edges in there correlates with the named numbers.

Reply:
Internally, the edge is saved as string, yes (`'Edge1'`) but from outside, that is, from the `make_function`, using `1` gives us a better way of converting this index to the internal representation. For example, if at some point in the future, the edges are named like `'EdgeA'`, then providing the string `'1'` or `'Edge1'` is not very useful anymore. So using the single integer `1` is better.

Yes, internally you should be extracting the edge from the `Edge[]` list always, whether that is a Curve or Spline or another thing, that is not important. The correlation is simple, `1` to `'Edge1'`, `2` to `'Edge2'`, etc. When you use the `getSelectionEx` function, it should return the proper string, which you can parse to get the number, this is done in several commands that pick things.
```py
sel = Gui.Selection.getSelectionEx()
sel_object = sel[0]
index = sel_object.SubElementNames[0].lstrip("Edge")  # this would be 1, 2, 3, or whichever

make_polar_array(..., axis_object="Line", axis_edge=index, ...)
```
This `index` should give you the corresponding edge.
 ```py
edge = sel_object.Object.Shape.Edges[index - 1]
# The list starts from 0 so we need to subtract 1
```

Reply:
For DatumLine this example won't work, is this a single exception?

```Python
>>> Gui.Selection.getSelectionEx()[0].SubElementNames
('Line',)
```
Can I always expect the method above to work in all cases where there is more than one edge and for my problematic example case just assume using the default first edge when there is no integer (as string) returned? Something like:

```
try:
   index = int(index)
except ValueError:
   index = 1
edge = sel_object.Object.Shape.Edges[index - 1]
```

Reply:
Is `index` supposed to be a string here?

I guess, yes, if the object only has one edge, we can use that one, like the case of the `DatumLine`.

Reply:
Yes the `index` should be considered to be a string in the example. I'll make the changes in the next commit then.